### PR TITLE
Jit_SystemRegisters: Make mfspr PIE-compliant

### PR DESCRIPTION
**Pull request by MerryMage**
Using RCX as a base register for accessing CoreTiming globals.

### CODE REVIEW

File: Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp [hunk 1]
```diff
@@ -282,18 +282,33 @@ void Jit64::mfspr(UGeckoInstruction inst)
     // no register choice
 
     gpr.FlushLockX(RDX, RAX);
+    gpr.FlushLockX(RCX);
+
+    const void* rcx_relative_base = &CoreTiming::g_last_OC_factor_inverted;
+    MOV(64, R(RCX), ImmPtr(rcx_relative_base));
+
+    // We're relying on these globals being placed close (within +/-2GB) to each other in memory.
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp]: ;_;

File: Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp [hunk 2]
```diff
@@ -282,18 +282,33 @@ void Jit64::mfspr(UGeckoInstruction inst)
...
+    const auto MRelative = [rcx_relative_base](void* ptr) -> OpArg {
+      const std::intptr_t base = reinterpret_cast<std::intptr_t>(rcx_relative_base);
+      const std::intptr_t target = reinterpret_cast<std::intptr_t>(ptr);
+      const std::ptrdiff_t offset = target - base;
+
+      _assert_(offset >= -0x80000000ll && offset <= 0x7FFFFFFFll);
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp]: `LL`

File: Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp [hunk 3]
```diff
@@ -282,18 +282,33 @@ void Jit64::mfspr(UGeckoInstruction inst)
     // no register choice
 
     gpr.FlushLockX(RDX, RAX);
+    gpr.FlushLockX(RCX);
+
+    const void* rcx_relative_base = &CoreTiming::g_last_OC_factor_inverted;
+    MOV(64, R(RCX), ImmPtr(rcx_relative_base));
+
+    // We're relying on these globals being placed close (within +/-2GB) to each other in memory.
+    const auto MRelative = [rcx_relative_base](void* ptr) -> OpArg {
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp]: I'd imagine the lambda return type would be inferred in this case, would it not?

File: Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp [hunk 4]
```diff
@@ -282,18 +282,33 @@ void Jit64::mfspr(UGeckoInstruction inst)
     // no register choice
 
     gpr.FlushLockX(RDX, RAX);
+    gpr.FlushLockX(RCX);
+
+    const void* rcx_relative_base = &CoreTiming::g_last_OC_factor_inverted;
+    MOV(64, R(RCX), ImmPtr(rcx_relative_base));
+
+    // We're relying on these globals being placed close (within +/-2GB) to each other in memory.
+    const auto MRelative = [rcx_relative_base](void* ptr) {
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp]: `ptr` here can be `const void*` can't it?

File: Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp (hunk 1 above)
Comment (degasus) [inline Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp]: What about moving all of those coretiming globals into a struct?

**Comment by MerryMage** [comment]
Naming the `struct` was difficult.

**Comment by degasus** [comment]
The JIT changes LGTM.

**Pull request merged**